app/uk/gov/hmrc/gform/sharedmodel/formtemplate/ComponentType.scala

```diff
@@ -158,16 +159,64 @@ object DisplayWidth extends Enumeration {
   implicit val displayWidthWrites: Writes[DisplayWidth] = Writes.enumNameWrites
 }
 
+sealed trait OptionData extends Product with Serializable
+
+object OptionData {
+
+  case class IndexBased(
+    label: SmartString
+  ) extends OptionData
+
+  case class ValueBased(
+    label: SmartString,
+    value: String
+  ) extends OptionData
+
+  private val templateReads: Reads[OptionData] = {
+
+    val indexBasedReads: Reads[OptionData] = Json.reads[IndexBased].widen[OptionData]
+    val valueBasedReads: Reads[OptionData] = Json.reads[ValueBased].widen[OptionData]
+
+    valueBasedReads | indexBasedReads
+  }
+
+  implicit val format: OFormat[OptionData] = OFormatWithTemplateReadFallback(templateReads)
+
+  implicit val leafExprs: LeafExpr[OptionData] = (path: TemplatePath, t: OptionData) =>
+    t match {
+      case OptionData.IndexBased(label)    => LeafExpr(path + "label", label)
+      case OptionData.ValueBased(label, _) => LeafExpr(path + "label", label)
+    }
+}
+
+sealed trait NoneChoice extends Product with Serializable
+
+object NoneChoice {
+
+  case class IndexBased(index: Int) extends NoneChoice
+  case class ValueBased(value: String) extends NoneChoice
+
+  private val templateReads: Reads[NoneChoice] = {
+
+    val indexBasedReads: Reads[NoneChoice] = Reads.IntReads.map(IndexBased.apply).widen[NoneChoice]
```

Comment:
should it be consistent with implementation of reads for OptionData templateReads?
```
    val indexBasedReads: Reads[NoneChoice] = Json.reads[IndexBased].widen[NoneChoice]
    val valueBasedReads: Reads[NoneChoice] = Json.reads[ValueBased].widen[NoneChoice]

```